python/src/spark_rapids_ml/utils.py

```diff
@@ -163,6 +163,14 @@ def _get_gpu_id(task_context: TaskContext) -> int:
     return gpu_id
 
 
+# When changing default rmm memory resources we retain the old ones
+# in this global array singleton to so that any (C++) allocations using them can
+# invoke the corresponding deallocate methods.  They will get cleaned up only when
```

Comment:
Does that cause the memory already allocated within the default RMM mem resources not to be freed?